lib/api/api.dart

```diff
@@ -779,4 +793,10 @@ class ApiSingleton {
       return false;
     }
   }
+  
+  Future<String> getCityNameFromCoords(double lat, double lon) async {
+    var locale = '${await UserManager.getLanguage()}_${await UserManager.getLanguageCountry()}';
+    var placemarks = await placemarkFromCoordinates(lat, lon, localeIdentifier: locale);
+    return placemarks[0].locality ?? '';
```

Comment:
If more than one placemark is returned, all of them will be very close to each other, so I'd just take the first one

Reply:
Need to check if placemarks is empty before getting the first position:
```dart
if (placemarks.isNotEmpty) {
    // code here
 }
 ```

Reply:
"locality" can be null, for example if it's in the middle of the ocean (0, 0). In this case I assume in the app it will just not show anything. I can modify the method signature to be able to return null values:
`Future<String?> getCityNameFromCoords(double lat, double lon) async {`

Another option is to do something like this to ensure that always either the locality or an empty string is returned
```
Future<String> getCityNameFromCoords(double lat, double lon) async {
    var locale = '${await UserManager.getLanguage()}_${await UserManager.getLanguageCountry()}';
    var placemarks = await placemarkFromCoordinates(lat, lon, localeIdentifier: locale);
    if (placemarks.isEmpty) { return ''; }
    return placemarks.first.locality ?? '';
}
```

Reply:
Yeah, returning always String looks good to me